src/main/java/com/bookstore/controller/publisher/PublisherController.java

```diff
@@ -0,0 +1,110 @@
+package com.bookstore.controller.publisher;
+import com.bookstore.jpa.book.BookRepository;
+import com.bookstore.jpa.publisher.Publisher;
+import com.bookstore.jpa.publisher.PublisherRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+
+/**
+ * The Publisher Rest API Controller
+ * @author Muneeb Nasir
+ * @version 4806.1
+ */
+@Controller
+public class PublisherController {
+
+    @Autowired
+    private BookRepository repository;
+
+    @Autowired
+    private PublisherRepository publisherRepo;
+
+    /**
+     * Getter for all publishers in the Database
+     * @param model
+     * @return Thymeleaf Page
+     */
+    @GetMapping(path = "/AmazinBookStore-publishers")
+    public String addressBookView(Model model) {
```

Comment:
Method name should be updated to match what it does, "address" is not appropriate in this context, maybe something like retrieveAllBooksView()